src/script/media/BackgroundBlurrer/BackgroundBlurrer.ts

```diff
@@ -0,0 +1,181 @@
+  };
+
+  // Create a buffer to put three 2d clip space points in
+  buffers = {position: gl.createBuffer(), textcoord: gl.createBuffer()};
+  // Bind it to ARRAY_BUFFER (think of it as ARRAY_BUFFER = positionBuffer)
+  gl.bindBuffer(gl.ARRAY_BUFFER, buffers.position);
+  // Set a rectangle the same size as the image.
+  setRectangle(gl, 0, 0, width, height);
+
+  // provide texture coordinates for the rectangle.
+  gl.bindBuffer(gl.ARRAY_BUFFER, buffers.textcoord);
+  gl.bufferData(
+    gl.ARRAY_BUFFER,
+    // This represents the 6 points of the 2 triangles reprensenting the restangle we want to paint
+    new Float32Array([0.0, 0.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0, 1.0]),
+    gl.STATIC_DRAW,
+  );
+
+  return gl;
+}
+
+/**
+ * Will blur a single frame of the given video.
+ * It will apply the segmentation mask on the area of the image we do not want to blur
+ * @param segmentationResults results from the segmentation
+ * @param videoElement the video element that plays the video we want to blur
+ * @param gl the webgl context
+ * @param param3 the width and height of the video
+ */
+export function blurBackground(
+  segmentationResults: ImageSegmenterResult,
+  videoElement: HTMLVideoElement,
+  gl: WebGLRenderingContext,
+  {width, height}: VideoDimensions,
+) {
+  // Clear the canvas
+  gl.viewport(0, 0, width, height);
+  gl.clearColor(0, 0, 0, 0);
+  gl.clear(gl.COLOR_BUFFER_BIT);
+
+  // Tell it to use our program (pair of shaders)
+  gl.useProgram(program);
+
+  // Bind the position buffer.
+  gl.enableVertexAttribArray(locations.position);
+  gl.bindBuffer(gl.ARRAY_BUFFER, buffers.position);
+  // Tell the position attribute how to get data out of positionBuffer (ARRAY_BUFFER)
+  const size = 2; // 2 components per iteration
+  const type = gl.FLOAT; // the data is 32bit floats
+  const normalize = false; // don't normalize the data
+  const stride = 0; // 0 = move forward size * sizeof(type) each iteration to get the next position
+  const offset = 0; // start at the beginning of the buffer
+  gl.vertexAttribPointer(locations.position, size, type, normalize, stride, offset);
+
+  // bind the texcoord buffer.
+  gl.enableVertexAttribArray(locations.texcoord);
+  gl.bindBuffer(gl.ARRAY_BUFFER, buffers.textcoord);
+  gl.vertexAttribPointer(locations.texcoord, size, type, normalize, stride, offset);
+
+  // set the size of the image
+  gl.uniform2f(locations.textureSize, width, height);
+
+  // start with the original image
+  // Create a texture and put the image in it.
+  const originalImageTexture = createAndSetupTexture(gl);
+  gl.texImage2D(gl.TEXTURE_2D, 0, gl.RGB, gl.RGB, gl.UNSIGNED_BYTE, videoElement);
+  gl.activeTexture(gl.TEXTURE0);
+  gl.bindTexture(gl.TEXTURE_2D, originalImageTexture);
+
+  // Assiging the segmentation mask to the mask uniform (so that it's accessible to the shader)
+  const segmentationMask = segmentationResults.confidenceMasks?.[0].getAsWebGLTexture() ?? null;
```

Comment:
How about an extra `?` if we get an empty list? 
```suggestion
  const segmentationMask = segmentationResults.confidenceMasks?.[0]?.getAsWebGLTexture() ?? null;
```